mlflow/tensorflow/__init__.py

```diff
@@ -935,7 +936,12 @@ def _get_input_data_slice():
                         input_example_slice = None
                         if isinstance(input_training_data, np.ndarray):
                             input_example_slice = input_training_data[:INPUT_EXAMPLE_SAMPLE_ROWS]
-                        elif isinstance(input_training_data, tensorflow.data.Dataset):
+                        elif (
+                            isinstance(input_training_data, tensorflow.data.Dataset) and
+                            # TensorFlow < 2.1.0 does not include methods for converting
+                            # a tf.data.Dataset to a numpy array, such as `as_numpy_iterator()`
+                            Version(tensorflow.__version__) >=  Version("2.1.0")
+                        ):
```

Comment:
Previously, this failed on TensorFlow 2.1.0, but we were skipping the tests for < 2.6.0, even though these tests didn't need to be skipped